generator/internal/mixin_utils_test.cc

```diff
@@ -25,9 +28,26 @@ using ::google::protobuf::FileDescriptor;
 using ::google::protobuf::ServiceDescriptor;
 using ::testing::AllOf;
 using ::testing::Contains;
+using ::testing::ElementsAre;
 using ::testing::Eq;
+using ::testing::Field;
 using ::testing::NotNull;
-using ::testing::Optional;
+
+MATCHER_P(HasMethod, expected, "has method full name") {
+  return arg.get().full_name() == expected;
+}
+
+MATCHER_P3(HasHttpRuleBinding, body, path, pattern_case,
+           "has http rule binding") {
+  return is_http_rule_expected(arg, body, path, pattern_case);
+}
+
+MATCHER_P4(HasAdditionalBinding, idx, body, path, pattern_case,
+           "has additional binding") {
+  if (idx >= arg.additional_bindings().size()) return false;
+  return is_http_rule_expected(arg.additional_bindings(idx), body, path,
+                               pattern_case);
+}
 
```

Comment:
nit: I think these are unused and can be removed now

Reply:
Done.